src/main/java/com/gregtechceu/gtceu/api/capability/recipe/FluidRecipeCapability.java

```diff
@@ -69,7 +93,33 @@ public FluidIngredient copyWithModifier(FluidIngredient content, ContentModifier
     @Override
     public List<AbstractMapIngredient> convertToMapIngredient(Object obj) {
         List<AbstractMapIngredient> ingredients = new ObjectArrayList<>(1);
-        if (obj instanceof FluidIngredient ingredient) {
+        if (obj instanceof IntProviderFluidIngredient ingredient) {
+            for (FluidIngredient.Value value : ingredient.values) {
+                if (value instanceof FluidIngredient.TagValue tagValue) {
+                    ingredients.add(new MapFluidTagIngredient(tagValue.getTag()));
+                } else {
+                    Collection<Fluid> fluids = value.getFluids();
+                    for (Fluid fluid : fluids) {
+//                        ingredients.add(new MapFluidIngredient(
+//                                new FluidStack(fluid,
+//                                        ingredient.getAmount(),
+//                                        ingredient.getNbt())));
+                        if (ingredient.getCountProvider() == null) {
+                            ingredients.add(new MapFluidIngredient(
+                                    new FluidStack(fluid,
+                                            ingredient.getAmount(),
+                                            ingredient.getNbt())));
+                        } else {
+                            ingredients.add(new MapFluidIngredient(
+                                    new FluidStack(fluid,
+                                            ingredient.getCountProvider().sample(GTValues.RNG),
+                                            ingredient.getNbt())));
+                        }
+                    }
+                }
+            }
```

Comment:
instead of copypasting all of it, do what `ItemRecipeCapability` does, e.g. operate on the inner ingredient.
```suggestion
            return convertToMapIngredient(ingredient.getInner());
```